src/main/java/com/parkhomovsky/bookstore/service/implementation/OrderServiceImpl.java

```diff
@@ -0,0 +1,173 @@
+                .collect(Collectors.toList());
+        order.setOrderItems(null);
+        orderItems.forEach(orderItem -> orderItem.setOrder(order));
+        OrderDto orderDto = orderMapper.toDto(order);
+        orderDto.setOrderItems(orderItemDtos);
+        orderItemsRepository.saveAll(orderItems);
+        return orderDto;
+    }
+
+    @Override
+    public List<OrderDto> getAll(Pageable pageable, Authentication authentication) {
+        User currentUser = (User) authentication.getPrincipal();
+        List<Order> orders = orderRepository.findAllByUser(currentUser);
+        List<OrderItem> orderItems = getUserOrderItems(currentUser, orders);
+        return orderItemsToOrderDtos(orders, orderItems);
+    }
+
+    @Override
+    public UpdateResponseDto updateStatus(Long orderId,
+                                          OrderUpdateStatusRequest updateStatusRequest) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Order with id "
+                        + orderId + " not found"));
+        Status newStatus = updateStatusRequest.getStatus();
+        order.setStatus(newStatus);
+        order = orderRepository.save(order);
+        return orderMapper.toUpdateResponse(order);
+    }
+
+    @Override
+    public List<OrderItemDto> getOrderItemsDto(Pageable pageable, Long orderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Order with id "
+                        + orderId + " not found"));
+        Long currentUserId = userService.getUserId();
+        List<OrderItem> orderItems = orderItemsRepository.findByOrder(order, currentUserId);
+        return orderItemsToOrderItemsDto(orderItems);
+    }
+
+    @Override
+    public OrderItemDto getOrderItemByidDto(Long orderId, Long itemId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Order with id "
+                        + orderId + " not found"));
+        Long currentUserId = userService.getUserId();
+        Optional<OrderItem> orderItem =
+                orderItemsRepository.findByIdAndOrder(itemId, order, currentUserId);
+        return orderItemsMapper.toDto(orderItem.orElseThrow(() ->
+                new EntityNotFoundException("Order item with id "
+                        + itemId + " not found in order with id " + orderId)));
+    }
+
+    private List<OrderItem> getUserOrderItems(User currentUser, List<Order> orders) {
+        List<Long> orderIds = orders.stream()
+                .map(Order::getId)
+                .collect(Collectors.toList());
+        return orderItemsRepository
+                .findAllByOrderIdInAndUserId(orderIds, currentUser.getId());
+    }
+
+    private List<OrderDto> orderItemsToOrderDtos(List<Order> orders, List<OrderItem> orderItems) {
+        Map<Long, List<OrderItem>> orderItemMap = orderItems.stream()
```

Comment:
Are you sure you need to perform grouping in the Java code? It'll load your server. You can fetch it grouped from the DB